src/Functions/FunctionBinaryArithmetic.h

```diff
@@ -1200,4 +1211,228 @@ class FunctionBinaryArithmetic : public IFunction
     bool canBeExecutedOnDefaultArguments() const override { return valid_on_default_arguments; }
 };
 
+
+template <template <typename, typename> class Op, typename Name, bool valid_on_default_arguments = true>
+class FunctionBinaryArithmeticWithConstants : public FunctionBinaryArithmetic<Op, Name, valid_on_default_arguments>
+{
+public:
+    using Base = FunctionBinaryArithmetic<Op, Name, valid_on_default_arguments>;
+    using Monotonicity = typename Base::Monotonicity;
+
+    static FunctionPtr create(
+        const ColumnWithTypeAndName & left_,
+        const ColumnWithTypeAndName & right_,
+        const DataTypePtr & return_type_,
+        const Context & context)
+    {
+        return std::make_shared<FunctionBinaryArithmeticWithConstants>(left_, right_, return_type_, context);
+    }
+
+    FunctionBinaryArithmeticWithConstants(
+        const ColumnWithTypeAndName & left_,
+        const ColumnWithTypeAndName & right_,
+        const DataTypePtr & return_type_,
+        const Context & context_)
+        : Base(context_), left(left_), right(right_), return_type(return_type_)
+    {
+    }
+
+    void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result, size_t input_rows_count) const override
+    {
+        if (left.column && isColumnConst(*left.column) && arguments.size() == 1)
+        {
+            Block block_with_constant
+                = {{left.column->cloneResized(input_rows_count), left.type, left.name},
+                   block.getByPosition(arguments[0]),
+                   block.getByPosition(result)};
+            Base::executeImpl(block_with_constant, {0, 1}, 2, input_rows_count);
+            block.getByPosition(result) = block_with_constant.getByPosition(2);
+        }
+        else if (right.column && isColumnConst(*right.column) && arguments.size() == 1)
+        {
+            Block block_with_constant
+                = {block.getByPosition(arguments[0]),
+                   {right.column->cloneResized(input_rows_count), right.type, right.name},
+                   block.getByPosition(result)};
+            Base::executeImpl(block_with_constant, {0, 1}, 2, input_rows_count);
+            block.getByPosition(result) = block_with_constant.getByPosition(2);
+        }
+        else
+            Base::executeImpl(block, arguments, result, input_rows_count);
+    }
+
+    bool hasInformationAboutMonotonicity() const override
+    {
+        std::string_view name_ = Name::name;
+        if (name_ == "minus" || name_ == "plus" || name_ == "divide" || name_ == "intDiv")
+        {
+            return true;
+        }
+        return false;
+    }
+
+    Monotonicity getMonotonicityForRange(const IDataType &, const Field & left_point, const Field & right_point) const override
```

Comment:
Actually since all of this does not uses generic abstractions, looks like it is better to cover all functions that listed here.

Reply:
Which one is not covered?

Reply:
Actually only the following is covered
- `divide(var, const)`

All of the above does not:
- `divide(const, var)`
- `intDiv(var, const)`
- `intDiv(const, var)`
- `plus(var, const)`
- `plus(const, var)`
- `minus(var, const)`
- `minus(const, var)`

Reply:
I don't follow. What do you mean by `cover` here? `plus` and `minus` take effect and catch a bug already.

Reply:
>What do you mean by cover here?

Add primary key with these functions into the 01480_binary_operator_monotonicity.sql

Reply:
Ok. So it's "test coverage" you were talking about.